reading_comprehension/qanet_for_drop.py

```diff
@@ -0,0 +1,330 @@
+
+        self._text_field_embedder = text_field_embedder
+
+        self._embedding_proj_layer = torch.nn.Linear(text_embed_dim, encoding_in_dim)
+        self._highway_layer = Highway(encoding_in_dim, num_highway_layers)
+
+        self._encoding_proj_layer = torch.nn.Linear(encoding_in_dim, encoding_in_dim)
+        self._phrase_layer = phrase_layer
+
+        self._matrix_attention = matrix_attention_layer
+
+        self._modeling_proj_layer = torch.nn.Linear(encoding_out_dim * 4, modeling_in_dim)
+        self._modeling_layer = modeling_layer
+
+        self._span_start_predictor = torch.nn.Linear(modeling_out_dim * 2, 1)
+        self._span_end_predictor = torch.nn.Linear(modeling_out_dim * 2, 1)
+
+        self._passage_weights_predictor = torch.nn.Linear(modeling_out_dim, 1)
+        self._question_weights_predictor = torch.nn.Linear(encoding_out_dim, 1)
+
+        self._answer_type_predictor = FeedForward(modeling_out_dim + encoding_out_dim,
+                                                  activations=[Activation.by_name('relu')(),
+                                                               Activation.by_name('linear')()],
+                                                  hidden_dims=[modeling_out_dim, 3],
+                                                  num_layers=2,
+                                                  dropout=dropout_prob)
+        self._number_embedding_proj_layer = torch.nn.Linear(text_embed_dim, modeling_out_dim)
+        self._number_sign_predictor = FeedForward(modeling_out_dim * 2,
+                                                  activations=[Activation.by_name('relu')(),
+                                                               Activation.by_name('linear')()],
+                                                  hidden_dims=[modeling_out_dim, 3],
+                                                  num_layers=2,
+                                                  dropout=dropout_prob)
+        self._count_number_predictor = FeedForward(modeling_out_dim,
+                                                   activations=[Activation.by_name('relu')(),
+                                                                Activation.by_name('linear')()],
+                                                   hidden_dims=[modeling_out_dim, 10],
+                                                   num_layers=2,
+                                                   dropout=dropout_prob)
+
+        self._span_start_accuracy = CategoricalAccuracy()
+        self._span_end_accuracy = CategoricalAccuracy()
+        self._span_accuracy = BooleanAccuracy()
+        self._squad_metrics = SquadEmAndF1()
+        self._dropout = torch.nn.Dropout(p=dropout_prob)
+
+        initializer(self)
+
+    def forward(self,  # type: ignore
+                question: Dict[str, torch.LongTensor],
+                passage: Dict[str, torch.LongTensor],
+                numbers_in_passage: Dict[str, torch.LongTensor],
```

Comment:
I'd recommend having this be an `IndexField` instead of a `TextField`.  This way you just _point_ to the positions of the numbers, and it should be easier on the model.  I'll comment more on this below.